examples/complete-http/main.tf

```diff
@@ -160,7 +160,7 @@ data "null_data_source" "downloaded_package" {
 
 module "lambda_function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "~> 1.0"
+  version = "1.38"
```

Comment:
I think it was good to use ranges here as it was before (`~> 1.0`). Or does it break anything?

Reply:
ya, so because we bumped the min supported version of terraform to 0.13.x in the lambda module, if we don't pin it to 1.38, it breaks the checks here because its saying that it needs to also have a min version of 0.13.x for terraform. so since these were just examples and the examples still work, I thought it would be less disruptive to just pin this to the version before that change instead of bumping the min terraform here to 0.13.x

Reply:
I see! It is indeed less disruptive to do now but in the future (or in other modules) we may have to run a different version of Terraform for each example folder. It will be very handy.

Reply:
we can do that today, we just can't do it with `pre-commit`

Reply:
Great, let's do it.

I think the only thing we need to run in each examples folder is `terraform_validate` by pre-commit with minimum version defined in a directory.

Additionally, the complete set of `pre-commit` checks will be executed using the max version for the complete repository.

Right?

Reply:
yes - if the max version between the examples and the module itself are the same, then we can keep the `max` check as it is today which is just running `pre-commit run -a` over the whole repo with the max version of terraform supported

for the min check, we would need to do the following:
1. get a list of directories to check (ideally, dynamically so we don't need to update github action file) - root of module (`.`), sub-modules (`modules/*`), examples (`examples/*`)
2. for each directory found in step 1:
	- extract min version of terraform supported
	- install min version of terraform supported
	- execute validation `terraform init && terraform validate`
	
so docs, linting, and formatting will be run only on max version supported, and all of the different combinations of min/max versions of terraform will be supported for validation

Reply:
Perfect. One minor comment that it should execute validation via `pre-commit run .... terraform_validate` so that we use pre-commit for everything.